install/kubernetes/cilium/README.md

```diff
@@ -122,10 +122,11 @@ contributors across the globe, there is almost always someone available to help.
 | enableK8sEventHandover | bool | `false` | Configures the use of the KVStore to optimize Kubernetes event handling by mirroring it into the KVstore for reduced overhead in large clusters. |
 | enableXTSocketFallback | bool | `true` |  |
 | encryption.enabled | bool | `false` | Enable transparent network encryption. |
-| encryption.keyFile | string | `"keys"` | Name of the key file inside the Kubernetes secret configured via secretName. |
-| encryption.mountPath | string | `"/etc/ipsec"` | Path to mount the secret inside the Cilium pod. |
-| encryption.nodeEncryption | bool | `false` | Enable encryption for pure node to node traffic. |
-| encryption.secretName | string | `"cilium-ipsec-keys"` | Name of the Kubernetes secret containing the encryption keys. |
+| encryption.keyFile | string | `"keys"` | Name of the key file inside the Kubernetes secret configured via secretName. This option is only effective when encryption.type is set to ipsec. |
+| encryption.mountPath | string | `"/etc/ipsec"` | Path to mount the secret inside the Cilium pod. This option is only effective when encryption.type is set to ipsec. |
+| encryption.nodeEncryption | bool | `false` | Enable encryption for pure node to node traffic. This option is only effective when encryption.type is set to ipsec. |
+| encryption.secretName | string | `"cilium-ipsec-keys"` | Name of the Kubernetes secret containing the encryption keys. This option is only effective when encryption.type is set to ipsec. |
+| encryption.type | string | `"ipsec"` | Encryption method. Can be either ipsec or wireguard. |
```

Comment:
Since all these encryption options are for `ipsec` rather than for both `ipsec` and `wireguard`, would it make sense to instead have an `ipsec` block for holding them, rather than bundling them in `encryption`?

E.g. either `encryption.ipsec.keyFile` or `ipsec.keyFile`.

We could then mark the existing `encryption.keyFile` and friends as deprecated (so as not to break anything for now) and eventually remove them in a few releases.

Reasoning: I'm not a fan of keeping a set of options that only work if another option at the same level is set to a specific value. I feel like this does not convey the intent properly, because there is no explicit link between the two, except from the documentation / diving into the chart itself. Hence why I think it makes more sense for specific options to be moved to a dedicated lower-level.

Reply:
Yeah, I think that's a good shout :+1: 

What I would do is:
* duplicate `keyFile`, `mountPath`, `nodeEncryption` and `secretName` under `encryption.ipsec`
* mark the options under `encryption` as deprecated
* give the deprecated options precedence over the ones under `encryption.ipsec` (in case they are both set)

@gandro @brb thoughts?

Reply:
I really don't have a preference as to whether or not we should do this this. I leave that discussion up to people with more experience with Helm best practices.

However, if we do it, then I think we should keep `nodeEncryption` generic. Even if it is not supported by the Wireguard-based encryption right now, it will likely become supported by Cilium 1.10.

The `keyFile`/`mountPath`/`secretName` however can probably safely be moved to a `ipsec` section. Even if we add support in the charts to inject a custom Wireguard private key via K8s secret in the future, the private keys for Wireguard needs to be specified per-node (where ipsec has a single key valid for all nodes) and thus the Helm values structure for that would look different as they will for ipsec.

Reply:
@nbusseneau I would then propose to merge this PR as it is since this is a release blocker, and add the new flags/deprecate the old ones in a separate PR, what do you think?

Reply:
Sure!